### PR TITLE
Diagnoses crud

### PR DESCRIPTION
**Pull request by PandaGrom**
# Description

[CRUD for Diagnosis](https://trello.com/c/7LYhQQlf/53-crud-for-diagnoses)

## Type of change

- [ ] Bug fix
- [x] New feature 

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/api/v1/admin/diagnoses.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          Diagnosis.create!(params)
+        end
+
+        desc 'Update a specific diagnosis'
+        route_param :id do
+          put do
+            diagnosis = Diagnosis.find(params[:id])
+
+            diagnosis.update(params)
+            diagnosis
```
Comment (hannazspot) [inline app/api/v1/admin/diagnoses.rb]:
```
Diagnosis.find(params[:id]).tap do |diagnosis|
  diagnosis.update!(params)
end
```

pls read about methods with a bang and what's the difference between `update` & `update!` and about `.tap`

File: app/api/v1/admin/diagnoses.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
+module V1
+  module Admin
+    class Diagnoses < API
+      desc 'create a new diagnosis'
+      # params do
+      #   requires :name, type: String
+      #   requires :symptom, type: String
+      #   requires :medication_list, type: String
+      # end
```
Comment (hannazspot) [inline app/api/v1/admin/diagnoses.rb]: remove comments pls

File: app/api/v1/diagnoses.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+module V1
+  class Diagnoses < API
```
Comment (hannazspot) [inline app/api/v1/diagnoses.rb:2]: why is a model's name in a plural form?

**Review by hannazspot — CHANGES_REQUESTED** [review]
pls add me in reviewers, thanks

File: spec/api/v1/admin/diagnoses_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+describe V1::Admin::Diagnoses do
+  include AuthHelper
+
+  let!(:current_user) { create(:user) }
+  let!(:diagnoses) { create_list(:diagnosis, 2) }
+
+  describe 'POST' do
```
Comment (hannazspot) [inline spec/api/v1/admin/diagnoses_spec.rb]: `POST` is not accurate enough in that context, you can have a couple of handlers that manage post queries, so this description doesn't suit here

**Review by hannazspot — APPROVED** [review]

**Review by alhimick-prog — APPROVED** [review]

**Pull request merged**